modules/concurrency/include/hephaestus/concurrency/message_queue_consumer.h

```diff
@@ -20,7 +19,7 @@ template <typename T>
 class MessageQueueConsumer {
 public:
   using Callback = std::function<void(T&&)>;
-  [[nodiscard]] MessageQueueConsumer(Callback&& callback, std::optional<std::size_t> max_queue_size);
+  [[nodiscard]] MessageQueueConsumer(Callback&& callback, std::size_t max_queue_size);
```

Comment:
Note that this is a pretty subtle breaking API change. Consider this caller code:

```cpp
MessageQueueConsumer mqc{someCallback, {}};
```

Prior to this PR, `{}` would be an empty optional. After this PR, `{}` would mean `0`. Not sure if any client will be broken by this.